sources/packages/backend/apps/queue-consumers/src/processors/schedulers/cra-integration/cra-process-integration.scheduler.ts

```diff
@@ -0,0 +1,50 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { CRAPersonalVerificationService } from "@sims/integrations/cra-integration/cra-personal-verification.service";
+import { QueueService } from "@sims/services/queue";
+import { QueueNames } from "@sims/utilities";
+import { InjectLogger, LoggerService } from "@sims/utilities/logger";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import { CRAValidationResultQueueOutDTO } from "./models/cra-validation-result.dto";
+
+@Processor(QueueNames.CRAProcessIntegration)
+export class CRAProcessIntegrationScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.CRAProcessIntegration)
+    protected readonly schedulerQueue: Queue<void>,
+    protected readonly queueService: QueueService,
+    private readonly cra: CRAPersonalVerificationService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  /**
+   * Identifies all the student applications that have a pending
+   * income verification and generate the request file to be
+   * processed by CRA.
+   * @params job job details.
+   * @returns Processing result log.
+   */
+  @Process()
+  async processIncomeVerification(
+    job: Job<void>,
```

Comment:
I'm confused here. The job's return type is void but processor is returning CRAValidationResultQueueOutDTO.
Both should be the same right?

Reply:
mm... job<void> is the input parameter of the function, `CRAValidationResultQueueOutDTO` is the output. 
![image](https://user-images.githubusercontent.com/77353155/207461998-0679467b-3398-440e-861f-add10dd19732.png)

Reply:
Nevermind 😊